lib/API/VK/Device.cpp

```diff
@@ -556,41 +648,77 @@ class VKDevice : public offloadtest::Device {
     llvm::SmallVector<VkWriteDescriptorSet> WriteDescriptors;
     assert(IS.BufferViews.empty());
     llvm::SmallVector<VkDescriptorBufferInfo> RawBufferInfos;
+    llvm::SmallVector<VkDescriptorImageInfo> ImageInfos;
+
+    // This massively over-reserves the descriptor info structs, but prevents
+    // dumb things from happening if these get reallocated...
+    const uint32_t DescriptorCount = P.getDescriptorCount();
+    RawBufferInfos.reserve(DescriptorCount);
+    ImageInfos.reserve(DescriptorCount);
```

Comment:
If we need the memory to be fixed in place we just shouldn't use a data structure that's resizable at all. Something like a `std::unique_ptr<VkDescriptorBufferInfo[]>` would be better.